### PR TITLE
show (partial) abstract by default

### PR DESCRIPTION
**Pull request by Dinika**
User-Story: [INSPIR-1028](https://its.cern.ch/jira/browse/INSPIR-1028)
Screenshots:

Partical abstract

screenshot from 2018-07-09 14-56-28
screenshot from 2018-07-09 14-56-13

Signed-off-by: Dinika Saxena <dinika.saxena@cern.ch>

### CODE REVIEW

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  constructor(props) {
+    super(props);
+    this.toggleLengthOfAbstract = this.toggleLengthOfAbstract.bind(this);
+    this.getRenderedAbstract = this.getRenderedAbstract.bind(this);
+    this.state = {
+      isAbstractExpanded: false,
+    };
+  }
+
+  getRenderedAbstract() {
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx]: getPartialOrFullAbstract

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    let renderedAbstract = this.props.abstract;
+    const { isAbstractExpanded } = this.state;
+    const minAbstractLength = 150;
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx]: move it outside of the class as  `const PARTIAL_ABSTRACT_LENGTH =`

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    super(props);
+    this.toggleLengthOfAbstract = this.toggleLengthOfAbstract.bind(this);
+    this.getRenderedAbstract = this.getRenderedAbstract.bind(this);
+    this.state = {
+      isAbstractExpanded: false,
+    };
+  }
+
+  getRenderedAbstract() {
+    let renderedAbstract = this.props.abstract;
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx]: `const { abstract } = this.props.abstract`, also see how it changes below.

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    const { isAbstractExpanded } = this.state;
+    const minAbstractLength = 150;
+    if (!isAbstractExpanded) {
+      renderedAbstract = this.props.abstract.substring(0, minAbstractLength);
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx]: and then `return abstract.substring(...`

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    return renderedAbstract;
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx]: `return abstract;`

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    this.setState({
+      isAbstractExpanded: !this.state.isAbstractExpanded,
+    });
+  }
+
+  render() {
+    const { abstract } = this.props;
+    return (
+      abstract && (
+        <div className="__AbstractContainer__">
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx]: this class name should be as same as the component name.

File: ui-tests/tests/literatureSearch.test.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const { routes } = require('../utils/constants');
```
Comment (harunurhan) [inline ui-tests/tests/literatureSearch.test.js]: irrelevant.

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  getPartialOrFullAbstract() {
+    let { abstract } = this.props;
+    const { isAbstractExpanded } = this.state;
+    if (!isAbstractExpanded) {
+      abstract = abstract && abstract.substring(0, PARTIAL_ABSTRACT_LENGTH);
+    }
+    return abstract;
+  }
+
+  toggleAbstractContainerHeight() {
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.jsx]: `toggleExpandedAbstractDisplay`

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss [hunk 9]
```diff
@@ -0,0 +1,12 @@
+.__CollapsibleAbstract__ {
+  margin-top: 0.5rem;
+  .abstractContainer {
+    line-height: 1.2rem;
+    .__Latex__ .katex span {
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss]: why did you need this hack? if it's really necessary and not a breaking change for other components that use `Latex`, we should move it to `Latex` component.

Comment (Dinika) [reply src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss]: I tried doing that. However, it messes up some titles.
screenshot from 2018-07-05 21-51-38

Comment (harunurhan) [reply src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss]: why did you need this in the first place?

Comment (Dinika) [reply src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss]: To display the 'More/Less' text right next to the abstract. (So, the class is applied to the span in abstract container)

Comment (harunurhan) [reply src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss]: @drjova let's not put it inline if this is the only way, because I am sure this css will break some latex rendering.

**Comment by StellaCh** [comment]
- The links should say 'Show more' and 'Show less' otherwise it might be confusing for some users.
- In a previous screenshot, you had the 'More' link inline in the 2nd line of the abstract. Can you use that instead? It was more intuitive and easy to grasp, as now someone could get confused to see the abstract cut-off until they realise that there's a link underneath to expand it.

**Comment by harunurhan** [comment]
@StellaCh that is very tricky due to latex rendering on that part, it can be done but very likely to cause weird latex rendering, so that's why I'd suggested to put it on the second line. We could append `...` so that it doesn't look that cut-off.

File: src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss [hunk 10]
```diff
@@ -0,0 +1,9 @@
+.__CollapsibleAbstract__ {
+  margin-top: 0.5rem;
+  .abstractContainer {
+    line-height: 1.2rem;
+  }
+  a {
```
Comment (harunurhan) [inline src/literature/components/CollapsibleAbstract/CollapsibleAbstract.scss]: add a class for this, applying styles to a tag directly is dangerous.

**Comment by StellaCh** [comment]
There are several reasons why the 'Show more' link shouldn't be in a new line:

- Our priority is to use the space as efficiently as possible in order to show the most search results possible. Taking 1 line just to show the link is not a good use of space
- Scanning quickly on the left should show all the important paper information in one glance. Among the important info, is the link and that attracts more attention that it should -  our goal is to quickly show them what they're looking for so the 'Show more' link should be as discreet as possible. 

If it's not possible to add the link as part of the second line, maybe we can think of using the collapse component with ... ?

**Comment by StellaCh** [comment]
could the 2nd line be a bit bigger? ~30 characters more?

**Comment by harunurhan** [comment]
@StellaCh we can make it but this only one screen size so it will be different for all, so putting more character might results in 3rd line on smaller screens.

**Comment by StellaCh** [comment]
ok leave it like this for the moment and we'll test it in different screens
